### DISCUSSION TITLE
How to get session's current transaction mode

### DISCUSSION THREAD

**Discussion by mcrio** [ravendb/ravendb#16448] [Q&A] [answered]
Hi,

I'm trying to find a way to query the session's current transaction mode.

Searching but couldn't find anything apart from the functionality to set the transaction mode via `session.Advanced.SetTransactionMode()`.

Writing a reusable library and want the method to fail in case the session does not have the expected transaction mode set.

edit: same question for querying if atomic guards are enabled

**Comment by garayx** [comment, marked as answer]
you can cast the session and access the `TransactionMode` field.


```
                using (var session = store.OpenAsyncSession(new SessionOptions
                       {
                           TransactionMode = TransactionMode.ClusterWide
                       }))
                {
                    AsyncDocumentSession a = (AsyncDocumentSession)session;
                    var txMode = a.TransactionMode;
                }
```

**Reply by mcrio** [reply]
Thanks @garayx omg I was checking but not sure how I missed that one.. Thank you :)
